pkg/resources/database_commons.go

```diff
@@ -0,0 +1,348 @@
+			ValidateDiag: StringInSlice(sdk.AsStringList(sdk.AllTraceLevels), true),
+			DiffSuppress: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+				return strings.EqualFold(oldValue, newValue)
+			},
+		},
+		{
+			Name:         sdk.ObjectParameterMaxDataExtensionTimeInDays,
+			Type:         schema.TypeInt,
+			Description:  "Object parameter that specifies the maximum number of days for which Snowflake can extend the data retention period for tables in the database to prevent streams on the tables from becoming stale. For a detailed description of this parameter, see [MAX_DATA_EXTENSION_TIME_IN_DAYS](https://docs.snowflake.com/en/sql-reference/parameters.html#label-max-data-extension-time-in-days).",
+			ValidateDiag: validation.ToDiagFunc(validation.IntBetween(0, 90)),
+		},
+		{
+			Name:        sdk.ObjectParameterReplaceInvalidCharacters,
+			Type:        schema.TypeBool,
+			Description: "Specifies whether to replace invalid UTF-8 characters with the Unicode replacement character (�) in query results for an Iceberg table. You can only set this parameter for tables that use an external Iceberg catalog.",
+		},
+		{
+			Name:         sdk.ObjectParameterStorageSerializationPolicy,
+			Type:         schema.TypeString,
+			Description:  fmt.Sprintf("The storage serialization policy for Iceberg tables that use Snowflake as the catalog. Valid options are: %v. COMPATIBLE: Snowflake performs encoding and compression of data files that ensures interoperability with third-party compute engines. OPTIMIZED: Snowflake performs encoding and compression of data files that ensures the best table performance within Snowflake.", sdk.AsStringList(sdk.AllStorageSerializationPolicies)),
+			ValidateDiag: StringInSlice(sdk.AsStringList(sdk.AllStorageSerializationPolicies), true),
+			DiffSuppress: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+				return strings.EqualFold(oldValue, newValue)
+			},
+		},
+		{
+			Name:         sdk.ObjectParameterSuspendTaskAfterNumFailures,
+			Type:         schema.TypeInt,
+			Description:  "How many times a task must fail in a row before it is automatically suspended. 0 disables auto-suspending.",
+			ValidateDiag: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+		{
+			Name:         sdk.ObjectParameterTaskAutoRetryAttempts,
+			Type:         schema.TypeInt,
+			Description:  "Maximum automatic retries allowed for a user task.",
+			ValidateDiag: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+		{
+			Name:        sdk.ObjectParameterUserTaskManagedInitialWarehouseSize,
+			Type:        schema.TypeString,
+			Description: "The initial size of warehouse to use for managed warehouses in the absence of history.",
+			DiffSuppress: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
tjhere is a nice function to do that now

Reply:
Used validation and diff suppress